packages/commonwealth/client/scripts/views/components/component_kit/polls/index.tsx

```diff
@@ -0,0 +1,243 @@
+/* @jsx m */
+
+import m from 'mithril';
+import moment from 'moment';
+import _ from 'lodash';
+import 'components/poll_card.scss';
+import { CWCard } from '../cw_card';
+import { CWRadioGroup } from '../cw_radio_group';
+import { CWButton } from '../cw_button';
+import { CWProgressBar, CWProgressBarStatus } from '../cw_progress_bar';
+import { CWIcon } from '../cw_icons/cw_icon';
+import { CWCheckbox } from '../cw_checkbox';
+import { CWRadioButton } from '../cw_radio_button';
+
+// Extend as use cases expand.
+export const enum PollType {
+  Offchain = 'Offchain',
+  Snapshot = 'Snapshot',
+}
+
+export const enum PollOptionResult {}
+
+export type VoteInformation = {
+  label: string;
+  value: string;
+  voteCount: number;
+};
+
+export type PollCardAttrs = {
+  pollType: PollType;
+  multiSelect: boolean;
+  pollEnded: boolean;
+  hasVoted: boolean;
+  votedFor: string;
+  proposalTitle: string;
+  timeRemainingString: string;
+  totalVoteCount: number;
+  voteInformation: VoteInformation[];
+  onVoteCast: () => any;
+  disableVoteButton: boolean;
+  incrementalVoteCast: number;
+  tokenSymbol?: string;
+};
+
+export function buildVoteDirectionString(voteOption: string) {
+  return `You voted "${voteOption}"`;
+}
+export class PollCard implements m.ClassComponent<PollCardAttrs> {
+  hasVoted: boolean; // keep
+  voteDirectionString: string; // keep
+  totalVoteCount: number; // keep
+  selectedOptions: string[];
+
+  oninit(vnode) {
+    // Initialize state which can change during the lifecycle of the component.
+    this.hasVoted = vnode.attrs.hasVoted;
+    this.voteDirectionString = vnode.attrs.votedFor
+      ? buildVoteDirectionString(vnode.attrs.votedFor)
+      : '';
+    this.totalVoteCount = vnode.attrs.totalVoteCount;
+    this.selectedOptions = [];
+  }
+
+  view(vnode) {
+    const {
+      pollType,
+      multiSelect,
+      pollEnded,
+      proposalTitle,
+      votedFor,
+      timeRemainingString,
+      voteInformation,
+      onVoteCast,
+      disableVoteButton = false,
+      tokenSymbol,
+      incrementalVoteCast,
+    } = vnode.attrs;
+
+    let resultString;
+    if (pollType === PollType.Offchain) {
+      resultString = 'Results'; // TODO: handle this and other poll types.
+    }
+
+    return (
+      <CWCard elevation="elevation-1" className="poll-card">
+        <div className="PollCard">
+          <div className="poll-title-section">
+            <div className="title">{proposalTitle}</div>
+          </div>
+          {!pollEnded && (
+            <div className="poll-voting-section">
+              {!this.hasVoted ? (
+                <div>
```

Comment:
also, we should try to not have unclassed divs like this. It's best to give it a name so that others can identify the purpose of the div (`poll-options-container` being an acceptable naming convention).